client/boot/index.js

```diff
@@ -161,8 +160,39 @@ function boot() {
 
 	translatorJumpstart.init();
 
-	reduxStore = createReduxStore();
+	localforage.config( {
+		name: 'calypso',
+		storeName: 'calypso_store',
+		description: 'Calypso Storage'
+	} );
+	localforage.getItem( 'redux-state', ( error, initialState ) => {
+		let reduxStore;
+		if ( error ) {
+			debug( 'failed to load initial redux-store state', error );
+			reduxStore = createReduxStore();
+		} else {
+			debug( 'loading redux-store with initial state', initialState );
+			reduxStore = createReduxStore( initialState || {} );
+		}
```

Comment:
This looks like we are very closely mixing the error-handling with the normal logic when we wouldn't have to. I don't see any difference between the two branches based on the error…

```js
// using a default to remove the || {}
( error, initialState = {} ) => {
	if ( error ) {
		debug( 'failed' );
	} else {
		debug( 'succeeded' );
	}

	const reduxStore = createReduxStore( initialState );
}
```